Makefile

```diff
@@ -1,52 +1,38 @@
-PROJECT_ROOT := src/
 VERSION = 1.0.7
 
 .DEFAULT_GOAL := all
 
-# The resulting binaries map to the subproject names
-BINARIES = \
-	abireport
+clean:
+	rm -f abireport abireport-*.tar.xz man/abireport.1 man/abireport.1.html
+	rm -fr vendor
 
-include Makefile.gobuild
-
-_PKGS = \
-	libabi \
-	explode \
-	abireport \
-	abireport/cmd
-
-# We want to add compliance for all built binaries
-_CHECK_COMPLIANCE = $(addsuffix .compliant,$(_PKGS))
-
-# Build all binaries as static binary
-BINS = $(addsuffix .statbin,$(BINARIES))
-
-# Ensure our own code is compliant..
-compliant: $(_CHECK_COMPLIANCE)
-install: $(BINS)
+install: all
 	test -d $(DESTDIR)/usr/bin || install -D -d -m 00755 $(DESTDIR)/usr/bin; \
-	install -m 00755 bin/* $(DESTDIR)/usr/bin/.; \
+	install -m 00755 abireport $(DESTDIR)/usr/bin/.; \
 	test -d $(DESTDIR)/usr/share/man/man1 || install -D -d -m 00755 $(DESTDIR)/usr/share/man/man1; \
 	install -m 00644 man/*.1 $(DESTDIR)/usr/share/man/man1/.; \
 
-# Credit to swupd developers: https://github.com/clearlinux/swupd-client
-MANPAGES = \
-	man/abireport.1
-
-gen_docs:
-	for MANPAGE in $(MANPAGES); do \
-		ronn --roff < $${MANPAGE}.md > $${MANPAGE}; \
-		ronn --html < $${MANPAGE}.md > $${MANPAGE}.html; \
-	done
-
-ensure_modules:
-	@ ( \
-		git submodule init; \
-		git submodule update; \
+gen_docs: man/abireport.1.md
+	@ronn --roff < man/abireport.1.md > man/abireport.1
+	@ronn --html < man/abireport.1.md > man/abireport.1.html
+
+vendor:
+	@go mod vendor
+
+all: vendor
+	(cd abi-report && go build --buildmode=pie -mod=vendor -o ../abireport)
+
+dist: vendor gen_docs
+	@rm -f abireport-$(VERSION).tar.xz
+	@git tag -l | grep -q v$(VERSION) || (echo "tag v$(VERSION) not found"; exit 1)
+	$(eval TMP := $(shell mktemp -d))
+	@cp -r . $(TMP)/abireport-$(VERSION)
+	@( \
+		cd $(TMP)/abireport-$(VERSION); \
+		git reset --hard v$(VERSION) &>/dev/null; \
+		git clean -xf; \
```

Comment:
This command is deleting the generated `man/abireport.1` and `man/abireport.1.html`, because the `-x` flag ignores the .gitignore file. However, we *do* want to remove some generated content (like the `abireport` binary). Not sure what the easiest path forward is...

Reply:
Ah, maybe use multiple `-e` options? Like `git clean -xf -e man/abireport.1 -e man/abireport.1.html`.

Reply:
Ah right, well originally I was thinking to just leave the generated content as checked into the repo (I had accidentally created empty files in the previous commit). I think that's probably the right thing to do and go back to not have gen_docs as part of make_dist.

Reply:
Just used -e instead as that was quicker.